tasks/vagrant.py

```diff
@@ -5,6 +5,38 @@
 from .common import PopenTask, PopenException, FallibleTask, TaskException
 
 
+def with_vagrant(func):
+    def wrapper(self, *args, **kwargs):
+        try:
+            __setup_provision(self)
+        except TaskException as exc:
+            logging.critical('vagrant or provisioning failed')
+            raise exc
+        else:
+            func(self, *args, **kwargs)
+        finally:
+            if not self.no_destroy:
+                self.execute_subtask(
+                    VagrantCleanup(raise_on_err=False))
+
+    return wrapper
+
+
+def __setup_provision(task):
+    """
+    This tries to execute the provision twice due to
+    problems described in issue #20
+    """
+    try:
+        task.execute_subtask(VagrantUp(timeout=None))
+        task.execute_subtask(VagrantProvision(timeout=None))
+    except:
```

Comment:
@tomaskrizek not sure if you want to log the error here (as debug at least). up to you